lib/Dialect/ATen/Transforms/RecognizeKernelsPass.cpp

```diff
@@ -49,9 +58,15 @@ convertTorchArgType(StringRef sourceTorchType, StringRef targetTorchType,
   // Immutable tensor conversion.
   if (flag & KVC::kImmutableTensor) {
     // TODO: Support the kPromoteScalar flag.
-    if (sourceTorchType != "Tensor" || targetTorchType != "Tensor")
+    if (!isTorchTensorType(sourceTorchType) ||
```

Comment:
when can it happen that a KVC::kImmutableTensor is not a tensor? comment?

Reply:
In the event of a mismatch between what the target conversion is asking for (conversion flag) and what the source op is. In that case, the conversion can't happen and we should fail it. It might happen around kernel/version skew, bugs, etc.